src/aten/UnaryOps.cpp

```diff
@@ -1,24 +1,163 @@
 #include <ATen/ScalarOps.h>
+#include <ATen/XPUNativeFunctions.h>
 #include <ATen/core/Tensor.h>
 #include <ATen/native/TensorIterator.h>
 #include <torch/library.h>
 
 #include <aten/sycl/UnaryKernels.h>
 
 namespace at {
-namespace native {
-namespace xpu {
 
-Tensor& abs_out(const Tensor& self, Tensor& out) {
+Tensor XPUNativeFunctions::abs(const Tensor& self) {
```

Comment:
XPUNativeFunctions should be part of XPUNativeFunctions.cpp. Meanwhile, aten/native contains UnaryOps.cpp in the stock pytorch; why do we need to provide a UnaryOps.cpp under aten?

Reply:
Please follow CUDA.

Reply:
`XPUNativeFunctions`, it's auto code-gen stub for ATen registration. Code-gen generates declaration, and operator is hand-written.

We follow stock PyTorch auto code-gen solution. We can have specific yaml, like `aten/src/ATen/native/ts_native_functions.yaml`.

When we start in-tree, we can migrate to native_functions.yaml or can keep it either.

Reply:
Do you mean the https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/UnaryOps.h is generated automatically by code-gen? By the way, we are in-tree now but just at the early stage. Community maintainers may review the code.

Reply:
We use https://github.com/pytorch/pytorch/blob/main/torchgen/gen_backend_stubs.py (https://github.com/intel/torch-xpu-ops/blob/f4b39f645a0ba1bbb1f44bda9c2115a648710039/cmake/Codegen.cmake#L15) to generate `RegisterXPU.cpp` (XPU resigtration stubs) and `XPUNativeFunctions.hpp` (Hooks declarations), like,
![image](https://github.com/intel/torch-xpu-ops/assets/28250760/596c0651-bb36-4657-9d97-8bf8a2c63b39)
We follow an approach used in PyTorch stock.